docs/cookbook/api.rst

```diff
@@ -0,0 +1,106 @@
+How to add a custom model?
+==========================
+
+In some cases you may be needing to add new models to your application in order to cover unique business needs.
+The proccess of extending Sylius with new entities is simple and intuitive.
+
+As an example we will take a **Supplier entity**, which may be really useful for shop maintenance.
+
+Authenitication
+---------------
+
+Creating OAuth client:
+
+.. code-block:: bash
+
+    $ php app/console sylius:oauth-server:create-client --grant-type="password" --grant-type="refresh_token" --grant-type="token"
+
+It will give you such a response:
+
+.. code-block:: bash
+
+    A new client with public id XYZ, secret ABC has been added
+
+Run your application on a built-in server:
+
+.. code-block:: bash
+
+    $ php app/console server:start localhost:8000
+
+To obtain authorization token for the ``api@example.com`` user with password ``sylius-api`` run:
+
+.. code-block:: bash
+
+    $ curl http://localhost:8000/api/oauth/v2/token -d "client_id"=XYZ -d "client_secret"=ABC -d "grant_type"=password -d "username"=api@example.com -d "password"=sylius-api
+
+This will give you such a response:
+
+.. code-block:: bash
+
+    {"access_token":"DEF","expires_in":3600,"token_type":"bearer","scope":null,"refresh_token":"GHI"}
+
+Creating a new resource instance via API
+----------------------------------------
+
+Use the ``access_token`` in the request that will create a new Supplier (:doc:`that we were creating in another cookbook</cookbook/custom-model>`).
+
+.. code-block:: bash
+
+    $ curl -i -X POST -H "Content-Type: application/json" -H "Authorization: Bearer DEF" -d '{"name": "Example", "description": "Lorem ipsum", "enabled": true}' http://localhost:8000/api/suppliers/
+
+.. tip::
+
+    Read more about authorizing in API :doc:`here </api/authorization>`.
+
+Viewing a single resource instance via API
+------------------------------------------
+
+If you would like to see the newly created resource use this command with ``1`` as ``{id)``. Remember to use **the authorization token**.
```

Comment:
Newly created resource is returned with the previous command. This one presents details about some resource with id = 1. It can be misleading. 
`resource use this command with 'obejct id' as '{id)'`